### PR TITLE
Add Support for HTTP Proxies

### PR DESCRIPTION
**Pull request by zicklag**
Hey there! I'm your first :star:. :smile:

This handles the only problem I have with `cargo readme` and is something I considered making because of the doc link problem. Thanks for this!

Here's a PR that adds support for HTTP proxies ( which I have to use ). Uses the code that the cargo CLI uses to add support.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # This file is automatically @generated by Cargo.
 # It is not intended for manual editing.
+version = 3
```
Comment (msrd0) [inline Cargo.lock:3]: Lockfile version 3 was introduced in 1.53 which is not stable yet. Can you please remove this line?

**Review by msrd0 — CHANGES_REQUESTED** [review]
Thanks for the PR! Don't mind the CI failure, that one's definitely my copy-paste error.

File: Cargo.lock (hunk 1 above)
Comment (zicklag) [reply Cargo.lock]: Fixed!

**Review by msrd0 — APPROVED** [review]
Thanks!

**Comment by zicklag** [comment]
No problem. I'm switching my projects to use this right now. You've even got a container image already that I can use for CI. It's perfect.

**Comment by msrd0** [comment]
> You've even got a container image already that I can use for CI

I'm also running this in CI. Do note though that I don't think the way I use cargo right now respects your lock file, so the docs.rs-links might change without you doing anything. If you know how to use cargo properly please tell me!

**Comment by zicklag** [comment]
Nope, unfortunately I've never used cargo as a lib. :man_shrugging:

**Pull request merged**